src/Microsoft.Health.Fhir.Shared.Core/Features/Operations/SearchParameterState/SearchParameterStateUpdateHandler.cs

```diff
@@ -0,0 +1,184 @@
+            {
+                throw new UnauthorizedFhirActionException();
+            }
+
+            _resourceSearchParameterStatus = await _searchParameterStatusManager.GetAllSearchParameterStatus(cancellationToken);
+            Dictionary<SearchParameterStatus, List<string>> searchParametersToUpdate = ParseRequestForUpdate(request, out List<OperationOutcomeIssue> invalidSearchParameters);
+            foreach (var statusGroup in searchParametersToUpdate)
+            {
+                await _searchParameterStatusManager.UpdateSearchParameterStatusAsync(statusGroup.Value, statusGroup.Key, cancellationToken);
+            }
+
+            return CreateBundleResponse(searchParametersToUpdate, invalidSearchParameters);
+        }
+
+        private Dictionary<SearchParameterStatus, List<string>> ParseRequestForUpdate(SearchParameterStateUpdateRequest request, out List<OperationOutcomeIssue> invalidSearchParameters)
+        {
+            var searchParametersToUpdate = new Dictionary<SearchParameterStatus, List<string>>();
+            invalidSearchParameters = new List<OperationOutcomeIssue>();
+
+            foreach (var searchParameter in request.SearchParameters)
+            {
+                System.Uri uri = searchParameter.Item1;
+                SearchParameterStatus status = searchParameter.Item2;
+                ResourceSearchParameterStatus searchParameterInfo = _resourceSearchParameterStatus.FirstOrDefault(sp => sp.Uri.Equals(uri));
+                if (searchParameterInfo == null)
+                {
+                    invalidSearchParameters.Add(new OperationOutcomeIssue(
+                        OperationOutcomeConstants.IssueSeverity.Information,
+                        OperationOutcomeConstants.IssueType.NotFound,
+                        detailsText: string.Format(Core.Resources.SearchParameterNotFound, uri)));
+                }
+                else if (!(status.Equals(SearchParameterStatus.Supported) || status.Equals(SearchParameterStatus.Disabled)))
+                {
+                    invalidSearchParameters.Add(new OperationOutcomeIssue(
+                        OperationOutcomeConstants.IssueSeverity.Error,
+                        OperationOutcomeConstants.IssueType.Invalid,
+                        detailsText: string.Format(Core.Resources.InvalidUpdateStatus, status, uri)));
+                }
+                else if (searchParameterInfo.Status.Equals(SearchParameterStatus.Deleted) || searchParameterInfo.Status.Equals(SearchParameterStatus.Unsupported))
+                {
+                    invalidSearchParameters.Add(new OperationOutcomeIssue(
+                        OperationOutcomeConstants.IssueSeverity.Error,
+                        OperationOutcomeConstants.IssueType.NotSupported,
+                        detailsText: string.Format(Core.Resources.SearchParameterDeleted, uri)));
+                }
+                else
+                {
```

Comment:
Just a thought, make a new variable statusKey.  If status == disabled, statusKey = pendingDisabled.  Then use statusKey for all the following lines, which may remove one level of nested if statements.